react/private/FieldMessage/FieldMessage.js

```diff
@@ -45,13 +45,19 @@ export default class FieldMessage extends Component {
 
     if (message) {
       const { critical, positive, secondary, ...restMessageProps } = this.props.messageProps;
+      const isCritical = (valid === false && !secondary) || critical;
+      const isPositive = (valid === true && !secondary) || positive;
+      const isSecondary = typeof valid === 'undefined' || secondary;
 
       return (
         <Text
           {...restMessageProps}
-          critical={(valid === false && !secondary) || critical}
-          positive={(valid === true && !secondary) || positive}
-          secondary={typeof valid === 'undefined' || secondary}>
+          critical={isCritical}
+          positive={isPositive}
+          secondary={isSecondary}
+          {...(isCritical && !restMessageProps.role) && { role: 'alert' }}
```

Comment:
I haven't seen this use of spread before.  I assume if the boolean on the left is true, it will spread the property in the literal on the right?